File: SYP.Web/Modules/Setting/NumberTemplates/Editor/NumberTemplatesEditorColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.NumberTemplatesEditor")]
[BasedOnRow(typeof(NumberTemplatesRow), CheckNames = true)]
public class NumberTemplatesEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Prefix { get; set; }
    public string DateFormat { get; set; }
    public string Suffix { get; set; }
    public int Length { get; set; }
    public int LengthText { get; set; }
    public bool Active { get; set; }
    public int Type { get; set; }
    public int DepartmentId { get; set; }
    public int InsertUserId { get; set; }
    public DateTime InsertDate { get; set; }
    public int UpdateUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int TenantId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/NumberTemplatesEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { NumberTemplateType } from "../_Ext/NumberTemplateType";
import { NumberTemplatesRow } from "./NumberTemplatesRow";

export interface NumberTemplatesEditorColumns {
    Id: Column<NumberTemplatesRow>;
    Prefix: Column<NumberTemplatesRow>;
    DateFormat: Column<NumberTemplatesRow>;
    Suffix: Column<NumberTemplatesRow>;
    Length: Column<NumberTemplatesRow>;
    LengthText: Column<NumberTemplatesRow>;
    Active: Column<NumberTemplatesRow>;
    Type: Column<NumberTemplatesRow>;
    DepartmentId: Column<NumberTemplatesRow>;
    InsertUserId: Column<NumberTemplatesRow>;
    InsertDate: Column<NumberTemplatesRow>;
    UpdateUserId: Column<NumberTemplatesRow>;
    UpdateDate: Column<NumberTemplatesRow>;
    TenantId: Column<NumberTemplatesRow>;
}

export class NumberTemplatesEditorColumns extends ColumnsBase<NumberTemplatesRow> {
    static readonly columnsKey = 'Setting.NumberTemplatesEditor';
    static readonly Fields = fieldsProxy<NumberTemplatesEditorColumns>();
}

[NumberTemplateType]; // referenced types